### DISCUSSION TITLE
Restore Menu Settings from Program Files

### DISCUSSION THREAD

**Discussion by aWESomness0** [Open-Shell/Open-Shell-Menu#1827] [Q&A] [answered]
I recently had an issue on my computer and to fix it, I had to delete everything and re-install Windows. I was able to save all the files, but my programs were lost.

I've re-installed Open-Shell, but I want to get my menu settings back to the way it was before I uninstalled the program. I still have all the files from the "program files" folder for this software, is there any way that I can restore any part of my menu from that data?

**Comment by ge0rdi** [comment]
`Open-Shell` doesn't store any settings in `Program Files` folder.

Settings are stored in `HKEY_CURRENT_USER\SOFTWARE\OpenShell` registry key.

**Reply by aWESomness0** [reply]
That's unfortunate. Guess I'll just have to recreate it from scratch. Thanks for letting me know!

**Reply by aWESomness0** [reply, marked as answer]
Actually I realized that since I had cloned my entire file system, I might have had access to the original registry values too! 

After a bit of research and some cautious care, I was easily able to simply replace the current Open-Shell registry values with those from my original data* and it restored (nearly) all of the data back! The only thing I think I lost was the pinned programs but that's a breeze to restore compared to redoing what I did restore.

*(I didn't touch the ones in the subfolder of the OpenShell folder of the same name ("HKEY_CURRENT_USER\SOFTWARE\OpenShell\OpenShell") as after inspection I figured those were probably related to the current installation of Open-Shell like the version and such, not its settings)
